jerry-core/lit/lit-literal-storage.h

```diff
@@ -17,466 +18,15 @@
-   */
-  void
-  set_size (size_t size) /**< size in bytes */
-  {
-    JERRY_ASSERT (size == get_size ());
-  } /* set_size */
-
-  /**
-   * Offset and length of 'prev' field, in bits
-   */
-  static const uint32_t prev_field_pos = _fields_offset_begin + 12u;
-  static const uint32_t prev_field_width = rcs_cpointer_t::bit_field_width;
-
-  static const size_t _header_size = RCS_DYN_STORAGE_LENGTH_UNIT;
-  static const size_t _size = _header_size + sizeof (ecma_number_t);
-}; /* lit_number_record_t */
-
-/**
- * Literal storage
- *
- * Represents flexible storage for the literals. The following records could be created inside the storage:
- * - charset literal (lit_charset_record_t)
- * - magic string literal (lit_magic_record_t)
- * - number literal (lit_number_record_t)
- */
-class lit_literal_storage_t : public rcs_recordset_t
-{
-public:
-  enum
-  {
-    LIT_TYPE_FIRST = _first_type_id,
-
-    LIT_STR = LIT_TYPE_FIRST,
-    LIT_MAGIC_STR,
-    LIT_MAGIC_STR_EX,
-    LIT_NUMBER,
-
-    LIT_TYPE_LAST = LIT_NUMBER
-  };
-
-  lit_charset_record_t *create_charset_record (const lit_utf8_byte_t *, lit_utf8_size_t);
-  lit_magic_record_t *create_magic_record (lit_magic_string_id_t);
-  lit_magic_record_t *create_magic_record_ex (lit_magic_string_ex_id_t);
-  lit_number_record_t *create_number_record (ecma_number_t);
-
-  uint32_t count_literals (void);
-
-  void dump ();
-
-private:
-  virtual rcs_record_t *get_prev (rcs_record_t *);
-  virtual void set_prev (rcs_record_t *, rcs_record_t *);
-  virtual size_t get_record_size (rcs_record_t *);
-}; /* lit_literal_storage_t */
-
-#define LIT_STR_T (lit_literal_storage_t::LIT_STR)
-#define LIT_MAGIC_STR_T (lit_literal_storage_t::LIT_MAGIC_STR)
-#define LIT_MAGIC_STR_EX_T (lit_literal_storage_t::LIT_MAGIC_STR_EX)
-#define LIT_NUMBER_T (lit_literal_storage_t::LIT_NUMBER)
-
-#ifdef JERRY_ENABLE_SNAPSHOT
-/**
- * Map from literal identifiers to the literal offsets in snapshot (or reverse)
- */
-typedef struct
-{
-  lit_cpointer_t literal_id; /**< identifier of literal in literal storage */
-  uint32_t literal_offset; /**< offset of the literal in snapshot */
-} lit_mem_to_snapshot_id_map_entry_t;
-
-extern bool
-lit_dump_literals_for_snapshot (uint8_t *, size_t, size_t *, lit_mem_to_snapshot_id_map_entry_t **,
-                                uint32_t *, uint32_t *);
-extern bool lit_load_literals_from_snapshot (const uint8_t *, uint32_t, lit_mem_to_snapshot_id_map_entry_t **,
-                                             uint32_t *, bool);
-#endif /* JERRY_ENABLE_SNAPSHOT */
+extern rcs_record_t *lit_create_charset_record (rcs_record_set_t *, const lit_utf8_byte_t *, lit_utf8_size_t);
+extern rcs_record_t *lit_create_magic_record (rcs_record_set_t *, lit_magic_string_id_t);
+extern rcs_record_t *lit_create_magic_record_ex (rcs_record_set_t *, lit_magic_string_ex_id_t);
+extern rcs_record_t *lit_create_number_record (rcs_record_set_t *, ecma_number_t);
```

Comment:
Sorry, I haven't noticed it last time, lets rename these functions to:
`lit_create_charset_literal`
`lit_create_magic_literal`
`lit_create_magic_literal_ex`
`lit_create_number_literal`

Thus we use _literal_  to describe a storage entity in _lit_ component, and _record_ - in _rcs_ component.